code/game/machinery/autolathe.dm

```diff
@@ -47,45 +46,272 @@
 							"Imported"
 							)
 
+	var/output_direction = "center"
```

Comment:
Overall I just thought it would be easier to have it as a string. There isn't much point in using the standard directional system for byond since it needs to have the 4 cardinal directions + center, and the proc that gets the offset turf doesn't take direction into account anyway, it uses an X and Y offset meaning having it as a direction number would add an extra conversion in the TGUI (converting directional number to string) and wouldn't actually remove any conversions from the get_output_turf proc.

Reply:
There are helper procs for dir to string. I say stick to standard

Reply:
Yea but then that would need an extra check for centre ejection and would either need to be added to the TGUI bundle anyway or converted beforehand in which case they would need to be converted back to numbers in the ui_act proc and used as the offset direction anyway.

Reply:
You can use 0 for center, I ike agrees you should switch to the dir helpers. There is a helper to convert back from string to dir aswell.

Reply:
Ok, I'll figure something out